File: 2075_Decode_the_Slanted_Ciphertext/solution.c
char* decodeCiphertext(char* e, int r) {
    int n = strlen(e);
    if (n == 0 || r == 0) {
        char* out = (char*)malloc(1);
        out[0] = '\0';
        return out;
    }
    int c = n / r, k = 0;
    char* res = (char*)malloc((n + 1) * sizeof(char));
    for (int s = 0; s < c; s++) {
        int i = 0, j = s;
        while (i < r && j < c) {
            res[k++] = e[i * c + j];
            i++; j++;
        }
    }
    res[k] = '\0';
    while (k > 0 && res[k - 1] == ' ') res[--k] = '\0';
    return res;
}

File: 2075_Decode_the_Slanted_Ciphertext/solution.py
class Solution:
    def decodeCiphertext(self, e: str, r: int) -> str:
        if not e or r == 0: return ""
        c = len(e) // r
        m = [list(e[i * c : (i + 1) * c]) for i in range(r)]
        res = []
        for s in range(c):
            i, j = 0, s
            while i < r and j < c:
                res.append(m[i][j])
                i += 1
                j += 1
        return "".join(res).rstrip()
